### ISSUE TITLE
Add-on upgrade fail during NVDA upgrade

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#16837]
### Steps to reproduce:
* Open add-on store
* Update an add-on, in my case Character Information, but do not close the store
* NVDA upgrade proposal appears to offer NVDA 2024.3beta2
* Upgrade NVDA from 2024.3beta1 to 2024.3beta2

### Actual behavior:
After NVDA's upgrade is complete and after NVDA has restarted, Character Information add-on is not installed anymore, neither the old version, nor the new one.

After restart the log contains the following error message:
```
ERROR - addonHandler.initialize (22:24:14.102) - MainThread (20388):
The following add-ons should be installed, but are no longer present on disk: charinfo
```

### Expected behavior:
The new add-on should be installed. If it is not possible, the limitation should be advertised, or NVDA's upgrade should not be possible when add-ons are in pending install state.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2024.3beta1 upgraded to 2024.3beta2
#### Windows version:
Windows 10 2004 (AMD64) build 19041.388
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Not tested
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Not tested
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tested
